### ISSUE TITLE
Can't disable hit highlighting

### ISSUE THREAD

**Issue by themerk** [typesense/typesense-instantsearch-adapter#94]
I'm not using snippets or highlighting, so I want to disable them to save bandwidth (I'm using Typesense Cloud so paying by the GB, and snippets double the payload size).

I assumed I could just disable them like so:
```ts
additionalSearchParameters: {
  highlight_fields: '',
  highlight_full_fields: '',
},
```

Unfortunately that doesn't work, because the code below just defaults it back to the query_by parameter. Unless I'm going about this the wrong way, I'd suggest a simple change of using `??` instead of `||`.

https://github.com/typesense/typesense-instantsearch-adapter/blob/bdea2599f109f85007646a5cd379952c10efcde5/src/Configuration.js#L27-L30

The logic also looks a bit faulty around `collectionSpecificSearchParameters`. If _any_ other collection-specific parameters are set, then I think `additionalSearchParameters.highlightFullFields` will be overridden back to query_by. I can see this potentially being confusing, when the expected behaviour is that only the given collection-specific parameters will override the defaults in `additionalSearchParameters`

https://github.com/typesense/typesense-instantsearch-adapter/blob/bdea2599f109f85007646a5cd379952c10efcde5/src/Configuration.js#L40

Example:
```ts
// I want to highlight just the name on both indexes
additionalSearchParameters: {
  highlight_full_fields: 'name'
},

collectionSpecificSearchParameters: {
  products: {
    query_by: 'name,barcode'
    // highlight_full_fields will unexpectedly default back to the query_by value
  },
  articles: {
    query_by: 'name,author,description'
    // highlight_full_fields will unexpectedly default back to the query_by value
  }
},
```

**Comment by jasonbosco** [comment]
@themerk I've fixed this in `v2.4.1-0`. Could you give it a shot now?

**Comment by themerk** [comment]
Looks good now, thanks for that 👍
